components/bio-formats-tools/src/loci/formats/tools/ImageConverter.java

```diff
@@ -893,9 +933,19 @@ private long convertTilePlane(IFormatWriter writer, int index, int outputIndex,
         int tileY = yCoordinate + y * h;
         int tileWidth = x < nXTiles - 1 ? w : width - (w * x);
         int tileHeight = y < nYTiles - 1 ? h : height - (h * y);
+
+        tryPrecompressed = precompressed && FormatTools.canUsePrecompressedTiles(reader, writer, writer.getSeries(), writer.getResolution());
         byte[] buf = getTile(reader, writer.getResolution(),
           index, tileX, tileY, tileWidth, tileHeight);
 
+        // if we asked for precompressed tiles, but that wasn't possible,
+        // then log that decompression/recompression happened
+        // TODO: decide if an exception is better here?
+        if (precompressed && !tryPrecompressed) {
+          LOGGER.warn("Decompressed tile: series={}, resolution={}, x={}, y={}",
+            writer.getSeries(), writer.getResolution(), x, y);
```

Comment:
If the whole resolution needs to be decompressed/recompressed, this should now only warn once. If some of the tiles in the resolution work with `-precompressed` and some do not, it will still warn once for each tile that doesn't work. Mostly I'd expect that to happen with right and bottom edge tiles, which could be smaller and need to be padded to the correct size.